lib/schema/validate.js

```diff
@@ -122,22 +122,41 @@ const processFieldNames = ({ payload = {}, nesting = [], schema = {}, results =
     const value = payload[f];
 
     if (Array.isArray(value)) {
-      value.forEach(
-        v => v instanceof Object && processFieldNames({ payload: v, nesting: [f, 'properties'], schema, results, resourceName })
-      );
+      const updatedNestedPayloads = [];
+      value.forEach(v => {
+        if (v instanceof Object) {
+          const { payload: updatedNestedPayload } = processFieldNames({
+            payload: v,
+            nesting: [f, 'properties'],
+            schema,
+            results,
+            resourceName
+          });
+          updatedNestedPayloads.push(updatedNestedPayload);
+        } else {
+          updatedNestedPayloads.push(v);
+        }
+      });
+      payload = {
+        ...payload,
+        [f]: updatedNestedPayloads
+      };
     }
 
     if (value?.['constructor'] === Object) {
-      processFieldNames({ payload: value, nesting, schema, results, resourceName });
+      const { payload: updatedNestedPayload } = processFieldNames({ payload: value, nesting, schema, results, resourceName });
+      if (updatedNestedPayload !== value) {
+        payload = {
+          ...payload,
+          [f]: updatedNestedPayload
+        };
+      }
     }
 
     const normalizedField = normalizeAtField(f);
 
     if (normalizedField !== f) {
-      delete payload[f];
-    }
-
-    if (f?.startsWith('@odata')) {
+      payload = { ...payload };
       delete payload[f];
```

Comment:
Should not be deleting anything from `payload`. In general, should never use `delete` unless there's a good reason. Better to build up a new object from scratch instead and return that. Params, especially, should never mutated.